keps/2941-DRA/README.md

```diff
@@ -0,0 +1,729 @@
+
+### Background
+
+DRA has four APIs that are relevant for a Kueue:
+
+- ResourceClaims
+- ResourceClaimTemplates
+- DeviceClasses
+- ResourceSlices
+
+#### DRA Example
+
+The easiest way to test DRA is to use [dra example driver repository](https://github.com/kubernetes-sigs/dra-example-driver). Cloning that repo and running 
+`make setup-e2e` will create a Kind cluster with the DRA feature gate and install a mock dra driver. This does not use
+actual GPUs so it is perfect for a test environment for exploring Kueue and DRA integration.
+
+#### Workload Example
+
+An example workload that uses DRA:
+
+```yaml
+---
+
+apiVersion: resource.k8s.io/v1alpha3
+kind: ResourceClaimTemplate
+metadata:
+  namespace: gpu-test1
+  name: single-gpu
+spec:
+  spec:
+    devices:
+      requests:
+      - name: gpu
+        deviceClassName: gpu.example.com
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: gpu-test1
+  name: job0
+  labels:
+    app: job
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: ctr0
+        image: ubuntu:22.04
+        command: ["bash", "-c"]
+        args: ["export; sleep 9999"]
+        resources:
+          claims:
+          - name: gpu
+          requests:
+            cpu: 1
+            memory: "200Mi"
+      resourceClaims:
+      - name: gpu
+        resourceClaimTemplateName: single-gpu
+```
+
+#### Example Driver Cluster Resources
+
+The dra-example-driver creates a ResourceSlice for each node and a DeviceClass named `gpu.example.com` for the entire 
+cluster.
+
+##### ResourceSlices
+
+ResourceSlices are meant for communication between drivers and the control planes. These are not expected to be used for
+workloads.
+
+##### DeviceClasses
+
+Each driver creates a device class and every resource claim will reference the device class. The dra-example-driver has
+a simple device class named `gpu.example.com`. This will be the way to enforce quota limits.
+
+### Goals
+
+- Users can submit workloads using resource claims and Kueue can monitor the usage.
+- Admins can enforce the quota for number of devices for a given DeviceClass.
+
+
+### Non-Goals
+
+- We are limiting scope for DRA to structured parameters (beta in 1.32 and 1.33)
+  - Support for alpha features like DRADeviceTaints, DRAAdminAccess, DRAPrioritizedLists and DRAPartitionableDevices
+    will not be included.
+- This design does not work with Topology Aware Scheduling feature of Kueue. It is a significant amount of work, will be
+   addressed in the future with a separate body of work
+
+## Proposal
+
+This proposal is to extend the APIs for allowing workloads using DRA APIs to be tallied against quota management, 
+borrowing and preemptable scheduling. This includes modifying:
+1. The `ResourceFlavor` API to allow creating a mapping canonical name -> device class names list
```

Comment:
```suggestion
1. Introduce API to allow creating a mapping canonical name -> device class names list
```
since we migrate to use CRD for the mapping.